pkg/kube/kube_suite_test.go

```diff
@@ -1,13 +1,83 @@
 package kube_test
 
 import (
+	"math/rand"
 	"testing"
+	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+var testEnv *envtest.Environment
+var testClustername = "test-cluster"
+var cfg *rest.Config
+var k8sClient client.Client
+var scheme *apiruntime.Scheme
+var k kube.Kube
+var k8sManager ctrl.Manager
+
 func TestKube(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Kube Suite")
 }
+
+var _ = BeforeSuite(func(done Done) {
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{"../../manifests/crds"},
+	}
+
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+
+	scheme = kube.CreateScheme()
+
+	k8sManager, err = ctrl.NewManager(cfg, ctrl.Options{
+		ClientDisableCacheFor: []client.Object{
+			&wego.Application{},
+			&corev1.Namespace{},
+			&corev1.Secret{},
+		},
+		Scheme: scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		err = k8sManager.Start(ctrl.SetupSignalHandler())
+		Expect(err).ToNot(HaveOccurred())
+	}()
+
+	k8sClient = k8sManager.GetClient()
+	Expect(k8sClient).ToNot(BeNil())
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+var letterRunes = []rune("abcdefghijklmnopqrstuvwxyz1234567890")
+
+func randStringRunes(n int) string {
```

Comment:
how about just `randomString` instead of including runes in the name?

But, you could also just delegate to https://pkg.go.dev/k8s.io/apimachinery/pkg/util/rand#String ?